src/bot/meigen/mod.rs

```diff
@@ -0,0 +1,305 @@
+        Self { db }
+    }
+
+    async fn make(&self, author: String, content: String) -> Result<String> {
+        let strip = |s: &str| s.trim().replace('`', "");
+
+        let author = strip(&author);
+        let content = strip(&content);
+
+        let len = author.chars().count() + content.chars().count();
+        if len > MEIGEN_LENGTH_LIMIT {
+            return Ok(
+                "名言が長すぎます({len}文字)。{MEIGEN_LENGTH_LIMIT}文字以下にしてください。".into(),
+            );
+        }
+
+        let meigen = self.db.save(author, content).await?;
+        Ok(meigen.to_string())
+    }
+
+    async fn show(&self, id: MeigenId) -> Result<String> {
+        Ok(match self.db.load(id).await? {
+            Some(x) => x.to_string(),
+            None => format!("No.{id} を持つ名言は見つかりませんでした。"),
+        })
+    }
+
+    async fn status(&self) -> Result<String> {
+        let count = self
+            .db
+            .count()
+            .await
+            .context("Failed to fetch meigen count")?;
+
+        Ok(format!(
+            "```yaml
+ total_count: {count}
+ ```",
+        ))
+    }
+
+    async fn search(&self, opt: FindOptions<'_>) -> Result<String> {
+        let mut res = self.db.search(opt).await?;
+
+        if res.is_empty() {
+            return Ok("条件に合致する名言が見つかりませんでした".into());
+        }
+
+        res.sort_unstable_by_key(|x| x.id);
+
+        Ok(list(&res))
+    }
+
+    async fn delete(&self, caller: u64, id: MeigenId) -> Result<String> {
+        const KAWAEMON_DISCORD_USER_ID: u64 = 391857452360007680;
+        if caller != KAWAEMON_DISCORD_USER_ID {
+            return Ok("名言削除はかわえもんにしか出来ません".into());
+        }
+
+        Ok(if self.db.delete(id).await? {
+            "削除しました".into()
+        } else {
+            format!("No.{id} を持つ名言は見つかりませんでした。")
+        })
+    }
+
+    async fn gophersay(&self, id: MeigenId) -> Result<String> {
+        let meigen = self.db.load(id).await.context("failed to get meigen")?;
+
+        Ok(match meigen {
+            Some(meigen) => format_gopher(&meigen),
+            None => format!("No.{id} を持つ名言は見つかりませんでした。"),
+        })
+    }
+
+    async fn love(&self, caller: u64, id: MeigenId) -> Result<String> {
+        Ok(if self.db.append_loved_user(id, caller).await? {
+            "いいねしました".to_string()
+        } else {
+            "すでにいいねしています".to_string()
+        })
+    }
+
+    async fn unlove(&self, caller: u64, id: MeigenId) -> Result<String> {
+        Ok(if self.db.remove_loved_user(id, caller).await? {
+            "いいねを解除しました".to_string()
+        } else {
+            "いいねしていません".to_string()
+        })
+    }
+}
+
+fn list(m: &[Meigen]) -> String {
+    let mut res = String::new();
+    let mut chars = 0;
+
+    for m in m {
+        if chars > LIST_LENGTH_LIMIT {
+            res.insert_str(0, "結果が長すぎたため一部の名言は省略されました\n");
+            break;
+        }
+
+        let m = m.to_string();
+        chars += m.chars().count();
+        res += &m;
+        res += "\n";
+    }
```

Comment:
```suggestion
fn list(meigens: &[Meigen]) -> String {
    let mut res = String::new();
    let mut chars = 0;

    for meigen in meigens {
        if chars > LIST_LENGTH_LIMIT {
            res.insert_str(0, "結果が長すぎたため一部の名言は省略されました\n");
            break;
        }

        let meigen = meigen.to_string();
        chars += meigen.chars().count();
        res += &meigen;
        res += "\n";
    }
```
このスコープの大きさに対して一文字変数はよろしくないので